android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/vitalsigns/VitalSignsDetailsViewModel.kt

```diff
@@ -84,6 +86,45 @@ constructor(val patientRepository: PatientRepository, var dispatcher: Dispatcher
       patientVitalItem.spO2 = observationValueOrDefault(listObservationspO2)
       patientVitalItem.spO2Unit = listObservationspO2.valueQuantity.unit ?: ""
 
+      //      val listObservationBmi = patientRepository.fetchVitalSigns(patientId = patientId,
+      // "bmi")
+      //      patientVitalItem.bmi = observationValueOrDefault(listObservationBmi)
+      //      patientVitalItem.bmiUnit = listObservationBmi.valueQuantity.unit ?: ""
+
+      // Todo: confirm if BMI can be displayed from
+      //  Add-Vitals height/weight if its of same Units
+      if (patientVitalItem.bmi.isEmpty() &&
+          patientVitalItem.height.isNotEmpty() &&
+          patientVitalItem.weight.isNotEmpty() &&
+          patientVitalItem.height.toDouble() > 0 &&
+          patientVitalItem.weight.toDouble() > 0
+      ) {
+        if (patientVitalItem.weightUnit.equals("kg", true) &&
+            patientVitalItem.heightUnit.equals("cm", true)
+        ) {
+          patientVitalItem.bmi =
+            computeBMIViaMetricUnits(
+                patientVitalItem.height.toDouble(),
+                patientVitalItem.weight.toDouble()
```

Comment:
`fetchVitalSigns` method increasing the complexity of code i think it could be broken into multiple functions.

Reply:
updated